File: ima-backend/src/main/java/com/techsolution/ima_backend/services/AgreementService.java
package com.techsolution.ima_backend.services;

import com.techsolution.ima_backend.dtos.request.AgreementRequest;
import com.techsolution.ima_backend.dtos.response.AgreementResponse;

import java.util.List;

public interface AgreementService {
    AgreementResponse createAgreement(Long applicationId, AgreementRequest agreementRequest);
    AgreementResponse createAgreementByAdmin(Long applicationId, Long teacherId, AgreementRequest agreementRequest);
    AgreementResponse validateAgreement(Long agreementId);
    List<AgreementResponse> getAllAgreements();
    AgreementResponse getAgreementById(Long agreementId);
    AgreementResponse updateAgreement(Long agreementId, AgreementRequest agreementRequest);
    void deleteAgreement(Long agreementId);
}

File: ima-frontend-react/src/services/AgreementService.js
import securedAxiosInstance from "./SecuredAxiosInstance";

const REST_API_AGREEMENTS = '/agreements';

export const listAgreements = () => securedAxiosInstance.get(`${REST_API_AGREEMENTS}`);

export const getAgreement = (agreementId) => 
    securedAxiosInstance.get(`${REST_API_AGREEMENTS}/${agreementId}`);

export const validateAgreement = (agreementId) => 
    securedAxiosInstance.put(`${REST_API_AGREEMENTS}/${agreementId}`);

export const updateAgreement = (agreementId, agreementRequest) => 
    securedAxiosInstance.put(`${REST_API_AGREEMENTS}/${agreementId}`, agreementRequest);

export const deleteAgreement = (agreementId) => 
    securedAxiosInstance.delete(`${REST_API_AGREEMENTS}/${agreementId}`);

export const createAgreement = (applicationId, agreementRequest) => 
    securedAxiosInstance.post(
        REST_API_AGREEMENTS,
        agreementRequest,
        {
            params: {applicationId}
        }
    );

export const createAgreementForAdmin = (applicationId, teacherId, agreementRequest) => 
    securedAxiosInstance.post(
        REST_API_AGREEMENTS,
        agreementRequest,
        {
            params: {applicationId, teacherId}
        }
    );    